### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Aduda-Boaz**
In this PR, I have created additional classes for the classroom, rental, and books.
I have implemented the relationship between the given classes.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+class Book
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+    @person = nil
+  end
+
+  attr_accessor :title, :author, :rentals, :person
```
Comment (FrederickMih) [inline book.rb]: - [Optional] These getters and setters are instance variables. Most often variables are usually defined at the very first beginning of the class. You may fix it :hammer:

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'rental'
+
+class Book
```
Comment (FrederickMih) [inline book.rb:3]: - [x] You will require a method for adding rental in order to fulfill the requirements of this project.
- `Hint`: When adding this method, the book itself should be sent as a parameter to create the new rental achieving the "has-many" association.
- [ReadMore](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

File: person.rb [hunk 3]
```diff
@@ -1,19 +1,35 @@
...
+  def correct_name
+    @name
+  end
+
+  def add_rental(rental, book)
+    book.rentals.push(rental) unless book.rentals.include?(rental)
+    book.person = self
+    @rentals.push(rental)
+    @book.push(book)
+  end
```
Comment (FrederickMih) [inline person.rb:31]: - [x] The person itself should be sent as a parameter to create the new rental achieving the "has-many" association. What did you think of trying it out like this?:

```ruby
  def add_rental(book, date)
    Rental.new(date, self, book)
  end
```

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @Aduda-Boaz , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



# To  Highlights

- GitHub workflow has been followed :wave: 
- PR has a title and summary :+1: 
- Linters are OK :heavy_check_mark:  

## Required Changes ♻️

- [x] Kindly rectify this issue in your README file. The project name is not correct. Also, make sure that the License Link is not broken.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by arslanbisharat — APPROVED** [review]
Hi @Aduda-Boaz,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**